GrandCentralBoard/Widgets/Watch/GoogleCalendarWatchWidgetBuilder.swift

```diff
@@ -14,15 +14,23 @@ final class GoogleCalendarWatchWidgetBuilder : WidgetBuilding {
     let name = "googleCalendarWatch"
 
     func build(settings: AnyObject) throws -> Widget {
+        let timeSettings = try TimeSourceSettings.decode(settings)
+        let calendarSettings = try GoogleCalendarSourceSettings.decode(settings)
+
         let networkRequestManager = Manager()
+        let tokenProvider = GoogleTokenProvider(clientID: calendarSettings.clientID,
+                                                clientSecret: calendarSettings.clientSecret,
+                                                refreshToken: calendarSettings.refreshToken)
+        let apiDataProvider = GoogleAPIDataProvider(tokenProvider: tokenProvider, networkRequestManager: networkRequestManager)
+        let calendarDataProvider = GoogleCalendarDataProvider(dataProvider: apiDataProvider)
```

Comment:
Can you explain the idea behind this chain of providers? 

tokenProvider -> apiDataProvider -> calendarDataProvider - What are the responsibilities of these objects?